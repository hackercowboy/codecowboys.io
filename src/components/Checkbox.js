import React from 'react';
import PropTypes from 'prop-types';

import { Field } from 'formik';
import ErrorMessage from './ErrorMessage';

function Checkbox({
  name, disabled, children,
}) {
  return (
    <div className="checkbox">
      <div className="checkbox-wrapper">
        <Field
          type="checkbox"
          id={name}
          name={name}
          disabled={disabled}
        />
        <label htmlFor={name}>
          {children}
        </label>
      </div>
      <ErrorMessage name={name} />
    </div>
  );
}

Checkbox.propTypes = {
  name: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

Checkbox.defaultProps = {
  disabled: false,
};

export default Checkbox;
