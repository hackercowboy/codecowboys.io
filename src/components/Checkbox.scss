.checkbox {
  height: 50px;
  margin-bottom: 20px;
}

.checkbox-wrapper {
  align-items: center;
  display: flex;
  justify-content: center;
  
  input {
    cursor: pointer;
    height: 20px;
    margin-right: 10px;
    width: 20px;
  }
  
}
