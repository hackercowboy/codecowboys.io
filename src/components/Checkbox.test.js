import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from './Checkbox';

describe('<Checkbox/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Checkbox />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
