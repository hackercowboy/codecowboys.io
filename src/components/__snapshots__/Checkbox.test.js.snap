// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> should render 1`] = `
"<div className=\\"checkbox\\">
  <div className=\\"checkbox-wrapper\\">
    <Field type=\\"checkbox\\" id={[undefined]} name={[undefined]} disabled={false} />
    <label htmlFor={[undefined]} />
  </div>
  <ErrorMessage name={[undefined]} />
</div>"
`;
